src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -69,80 +67,112 @@ template Instruction
+            discloseA passthroughAccount
+            vSenderAccount <- view <$> fetchInterfaceByKey @Account.R passthroughAccount
+            assertMsg "actors must include instructors" $ S.isSubsetOf vSenderAccount.controllers.instructors actors
+            pure $ PassthroughFrom (passthroughAccount, fromInstruction)
           a -> do
             assertMsg "actors must include sender" $ S.member step.sender actors
             pure a
-        -- update instruction state
         cid <- toInterfaceContractId <$> create this with allocation = allocationDisclosed; signed = S.insert step.sender signed
         pure (cid, otCid)
-      approve Instruction.Approve{approval; actors} = do
+      approve Instruction.Approve{actors; approval} = do
+        let newSigned = S.insert step.receiver signed
         assertMsg ("approval must be new" <> messageSuffix) $ approval /= this.approval
-        -- clean up if previously approved
+        -- undisclose previous approval
         case this.approval of
           TakeDelivery receiverAccount -> undiscloseA receiverAccount
+          PassthroughTo (passthroughAccount, _) -> undiscloseA passthroughAccount
           a -> pure None
-        -- disclose approval
+        -- approve
         case approval of
           TakeDelivery receiverAccount -> do
-            vReceiverAccount <- view <$> fetchInterfaceByKey @Account.R receiverAccount
             discloseA receiverAccount
-            assertMsg "actors must include transfer approvers" $ S.isSubsetOf vReceiverAccount.controllers.approvers actors
+            vReceiverAccount <- view <$> fetchInterfaceByKey @Account.R receiverAccount
+            assertMsg "actors must include approvers" $ S.isSubsetOf vReceiverAccount.controllers.approvers actors
+          PassthroughTo (passthroughAccount, toInstruction) -> do
+            discloseA passthroughAccount
+            vReceiverAccount <- view <$> fetchInterfaceByKey @Account.R passthroughAccount
+            assertMsg "actors must include approvers" $ S.isSubsetOf vReceiverAccount.controllers.approvers actors
           _ ->
             assertMsg "actors must include receiver" $ S.member step.receiver actors
-        -- update instruction state
-        toInterfaceContractId <$> create this with signed = S.insert step.receiver signed; approval
-      execute Instruction.Execute{actors; undiscloseSenderAccount; undiscloseReceiverAccount} = do
+
+        toInterfaceContractId <$> create this with signed = newSigned; approval
+      execute Instruction.Execute{actors} = do
         assertMsg "actors must intersect with settlers" $ not $ S.null $ S.intersection actors settlers
+        let void allocation approval = abort $ "unsupported combination " <> show (allocation, approval) <> messageSuffix
         -- execute instruction
         case (allocation, approval) of
-          (Unallocated, _) -> abort $ "instruction must be allocated" <> messageSuffix
```

Comment:
This error message was more descriptive than "unsupported combination"

Reply:
agree, I made the following more descriptive:
```
(Unallocated, Unapproved) -> abort $ "instruction must be allocated and approved" <> messageSuffix
(Unallocated, _) -> abort $ "instruction must be allocated" <> messageSuffix
(_, Unapproved) -> abort $ "instruction must be approved" <> messageSuffix
```

Reply:
```
(PassthroughFrom _, _) -> abort $ "asset has not been passed through" <> messageSuffix
```

Reply:
added more descriptive abort messages for all cases